common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -405,19 +469,35 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
+                    result = DeadlineRunResult.TIMED_OUT;
```

Comment:
> should we check here if we have run all tasks? if we have run all tasks claiming that we TIMED_OUT seems misleading.

@NiteshKant 

> This is tricky as `TIMED_OUT` means that the deadline expired, based on which I take decisions in `SingleThreadEventLoop`. If I return `RAN_AT_LEAST_ONCE` here then the caller has to check whether the deadline expired or not. May be I can rename this to `DEADLINE_EXPIRED`?

Do we even need the return status any more? The `executionTimeCallback` is used by SingleThreadedEventLoop to determine if there is any execution time remaining, and this `TIMED_OUT` is only checked periodically anyways so it isn't reliable.